feat: add support for custom anomaly threshold

Ecalzo opened the pull request:
Aims to implement https://github.com/elementary-data/elementary/issues/71

* Introduces and implements `anomaly_threshold` arg in `get_anomaly_query_macros`
* Refactors logic to get the `anomaly_threshold` from user input (if provided) or config into a separate file under `test_utils/`
* Introduces the arg into `test_` files that use this function

Glad to incorporate any feedback received to help implement this 🚀 
Little fuzzy on how to run SQL integration tests, but happy to learn.

oravi commented:
Hi @Ecalzo ! thanks so much for opening this PR! the code looks good to me, great job here 💯 👍 
Regarding tests, I've actually written a guide for it today [here](https://docs.elementary-data.com/project-overview/contributions/dbt-package), I can also expedite this phase and run the tests for you but would also love to hear your feedback about it. Please let me know what you think and I'll get back to you with the test results of this branch in our environment

Ecalzo commented:
> Hi @Ecalzo ! thanks so much for opening this PR! the code looks good to me, great job here 💯 👍 Regarding tests, I've actually written a guide for it today [here](https://docs.elementary-data.com/project-overview/contributions/dbt-package), I can also expedite this phase and run the tests for you but would also love to hear your feedback about it. Please let me know what you think and I'll get back to you with the test results of this branch in our environment

Thank you @oravi :) Some quick feedback after setting up and running an initial test:
* Docs say to have a profile `elementary` in your profiles.yml file, but the `integration_tests/` folder has it's own profile that expects, `elementary_tests` for running tests.
* Needed to pip install `dbt-snowflake` connector in addition to elementary and run `dbt deps`
* Needed to create the following folders manually: `$ mkdir integration_tests/data/training` and `$ mkdir integration_tests/data/validation`

Current progress:
Running the e2e test command for tables currently errors for me, even with a clean version of  elementary-data/dbt-data-reliability so I am working to understand that a bit, but it seems that it's not my changes throwing these errors. @oravi do you know if there are any specific unit and e2e tests you'd prefer me to run to test these changes? This can help me focus debug efforts.

Here is an important snip of that error log (this is in a clean repository without my changes and a new Snowflake trial account with a newly created db). 

```shell
$ python ./run_e2e_tests.py -t snowflake -e table

12:45:52  Completed with 3 errors and 1 warning:
12:45:52
12:45:52  Failure in test elementary_source_table_anomalies_training_string_column_anomalies_training_ (models/schema.yml)
12:45:52    Got 2 results, configured to fail if != 0
12:45:52
12:45:52    compiled SQL at target/compiled/elementary_integration_tests/models/schema.yml/elementary_source_table_anomal_44af3aa8a3eb7c5836a88d2678957df2.sql
12:45:52
12:45:52  Failure in test elementary_table_anomalies_numeric_column_anomalies_ (models/schema.yml)
12:45:52    Got 2 results, configured to fail if != 0
12:45:52
12:45:52    compiled SQL at target/compiled/elementary_integration_tests/models/schema.yml/elementary_table_anomalies_numeric_column_anomalies_.sql
12:45:52
12:45:52  Failure in test elementary_table_anomalies_string_column_anomalies_freshness (models/schema.yml)
12:45:52    Got 1 result, configured to fail if != 0
12:45:52
12:45:52    compiled SQL at target/compiled/elementary_integration_tests/models/schema.yml/elementary_table_anomalies_string_column_anomalies_freshness.sql
12:45:52
12:45:52  Warning in test elementary_table_anomalies_any_type_column_anomalies_row_count (models/schema.yml)
12:45:52    Got 1 result, configured to warn if != 0
12:45:52
12:45:52    compiled SQL at target/compiled/elementary_integration_tests/models/schema.yml/elementary_table_anomalies_any_type_column_anomalies_row_count.sql
12:45:52
12:45:52  Done. PASS=1 WARN=1 ERROR=3 SKIP=0 TOTAL=5
2022-05-30 08:45:53,061 — monitor.dbt_runner — INFO — Running dbt --log-format json run-operation validate_table_anomalies --project-dir . --profiles-dir /home/ec/.dbt --target snowflake (this might take a while)
SUCCESS: ['STRING_COLUMN_ANOMALIES_TRAINING', 'STRING_COLUMN_ANOMALIES', 'NUMERIC_COLUMN_ANOMALIES'] in list ['string_column_anomalies', 'numeric_column_anomalies', 'string_column_anomalies_training']
SUCCESS: ['STRING_COLUMN_ANOMALIES_TRAINING', 'ANY_TYPE_COLUMN_ANOMALIES', 'NUMERIC_COLUMN_ANOMALIES'] in list ['any_type_column_anomalies', 'numeric_column_anomalies', 'string_column_anomalies_training']

snowflake results

snowflake results

Table test results
SUCCESS: ['STRING_COLUMN_ANOMALIES_TRAINING', 'STRING_COLUMN_ANOMALIES', 'NUMERIC_COLUMN_ANOMALIES'] in list ['string_column_anomalies', 'numeric_column_anomalies', 'string_column_anomalies_training']
SUCCESS: ['STRING_COLUMN_ANOMALIES_TRAINING', 'ANY_TYPE_COLUMN_ANOMALIES', 'NUMERIC_COLUMN_ANOMALIES'] in list ['any_type_column_anomalies', 'numeric_column_anomalies', 'string_column_anomalies_training']

String columns test results

Numeric columns test results

Any type columns test results

Schema changes test results

Regular test results

dbt artifacts results
```

oravi commented:
> > Hi @Ecalzo ! thanks so much for opening this PR! the code looks good to me, great job here 💯 👍 Regarding tests, I've actually written a guide for it today [here](https://docs.elementary-data.com/project-overview/contributions/dbt-package), I can also expedite this phase and run the tests for you but would also love to hear your feedback about it. Please let me know what you think and I'll get back to you with the test results of this branch in our environment
> 
> Thank you @oravi :) Some quick feedback after setting up and running an initial test:
> 
> * Docs say to have a profile `elementary` in your profiles.yml file, but the `integration_tests/` folder has it's own profile that expects, `elementary_tests` for running tests.
> * Needed to pip install `dbt-snowflake` connector in addition to elementary and run `dbt deps`
> * Needed to create the following folders manually: `$ mkdir integration_tests/data/training` and `$ mkdir integration_tests/data/validation`
> 
> Current progress: Running the e2e test command for tables currently errors for me, even with a clean version of elementary-data/dbt-data-reliability so I am working to understand that a bit, but it seems that it's not my changes throwing these errors. @oravi do you know if there are any specific unit and e2e tests you'd prefer me to run to test these changes? This can help me focus debug efforts.
> 
> Here is an important snip of that error log (this is in a clean repository without my changes and a new Snowflake trial account with a newly created db).
> 
> ```shell
> $ python ./run_e2e_tests.py -t snowflake -e table
> 
> 12:45:52  Completed with 3 errors and 1 warning:
> 12:45:52
> 12:45:52  Failure in test elementary_source_table_anomalies_training_string_column_anomalies_training_ (models/schema.yml)
> 12:45:52    Got 2 results, configured to fail if != 0
> 12:45:52
> 12:45:52    compiled SQL at target/compiled/elementary_integration_tests/models/schema.yml/elementary_source_table_anomal_44af3aa8a3eb7c5836a88d2678957df2.sql
> 12:45:52
> 12:45:52  Failure in test elementary_table_anomalies_numeric_column_anomalies_ (models/schema.yml)
> 12:45:52    Got 2 results, configured to fail if != 0
> 12:45:52
> 12:45:52    compiled SQL at target/compiled/elementary_integration_tests/models/schema.yml/elementary_table_anomalies_numeric_column_anomalies_.sql
> 12:45:52
> 12:45:52  Failure in test elementary_table_anomalies_string_column_anomalies_freshness (models/schema.yml)
> 12:45:52    Got 1 result, configured to fail if != 0
> 12:45:52
> 12:45:52    compiled SQL at target/compiled/elementary_integration_tests/models/schema.yml/elementary_table_anomalies_string_column_anomalies_freshness.sql
> 12:45:52
> 12:45:52  Warning in test elementary_table_anomalies_any_type_column_anomalies_row_count (models/schema.yml)
> 12:45:52    Got 1 result, configured to warn if != 0
> 12:45:52
> 12:45:52    compiled SQL at target/compiled/elementary_integration_tests/models/schema.yml/elementary_table_anomalies_any_type_column_anomalies_row_count.sql
> 12:45:52
> 12:45:52  Done. PASS=1 WARN=1 ERROR=3 SKIP=0 TOTAL=5
> 2022-05-30 08:45:53,061 — monitor.dbt_runner — INFO — Running dbt --log-format json run-operation validate_table_anomalies --project-dir . --profiles-dir /home/ec/.dbt --target snowflake (this might take a while)
> SUCCESS: ['STRING_COLUMN_ANOMALIES_TRAINING', 'STRING_COLUMN_ANOMALIES', 'NUMERIC_COLUMN_ANOMALIES'] in list ['string_column_anomalies', 'numeric_column_anomalies', 'string_column_anomalies_training']
> SUCCESS: ['STRING_COLUMN_ANOMALIES_TRAINING', 'ANY_TYPE_COLUMN_ANOMALIES', 'NUMERIC_COLUMN_ANOMALIES'] in list ['any_type_column_anomalies', 'numeric_column_anomalies', 'string_column_anomalies_training']
> 
> snowflake results
> 
> snowflake results
> 
> Table test results
> SUCCESS: ['STRING_COLUMN_ANOMALIES_TRAINING', 'STRING_COLUMN_ANOMALIES', 'NUMERIC_COLUMN_ANOMALIES'] in list ['string_column_anomalies', 'numeric_column_anomalies', 'string_column_anomalies_training']
> SUCCESS: ['STRING_COLUMN_ANOMALIES_TRAINING', 'ANY_TYPE_COLUMN_ANOMALIES', 'NUMERIC_COLUMN_ANOMALIES'] in list ['any_type_column_anomalies', 'numeric_column_anomalies', 'string_column_anomalies_training']
> 
> String columns test results
> 
> Numeric columns test results
> 
> Any type columns test results
> 
> Schema changes test results
> 
> Regular test results
> 
> dbt artifacts results
> ```

Thanks so much for the feedback @Ecalzo ! 
* Regarding the requirements - you are completely right and I'll add all of these to the docs as well (my environment was already configured with all of these so I missed it, sorry about it)
* Regarding the error - actually I think that it's working for you! as you can see under table test results there are SUCCESS messages which makes sense as you ran only the table e2e tests. The output could be definitely better and more indicative so I'll improve it down the road as well.

oravi commented:
@Ecalzo I've just finished running the tests (both unit and e2e tests) for this branch and everything is working correctly. 

One small thing that came up while discussing about this with other users is that maybe the name 'anomaly_threshold' is not clear when referring to anomalies (specifically the word threshold). What do you think about exposing the 'anomaly_threshold' test param as 'sensitivity'? (all internal macros could remain the same). Would love to hear your opinion about it as well and really sorry for bringing it up so late in the process

Ecalzo commented on models/edr/data_monitoring/anomaly_detection/metrics_anomaly_score.sql:
```diff
@@ -70,7 +70,7 @@ final as (
     select *,
         case
-            when abs(z_score) > {{ elementary.get_config_var('anomaly_score_threshold') }} then true
+            when abs(z_score) > {{ elementary.get_config_var('anomaly_sensitivity') }} then true
```
Since we are changing the config var name, I've needed to update the name in a previously untouched file

Ecalzo commented:
> @Ecalzo I've just finished running the tests (both unit and e2e tests) for this branch and everything is working correctly.
> 
> One small thing that came up while discussing about this with other users is that maybe the name 'anomaly_threshold' is not clear when referring to anomalies (specifically the word threshold). What do you think about exposing the 'anomaly_threshold' test param as 'sensitivity'? (all internal macros could remain the same). Would love to hear your opinion about it as well and really sorry for bringing it up so late in the process

@oravi I've rebased and changed the name, please take another look when you have a moment. I ran the e2e tests, here is the final output:

```shell
$ python run_e2e_tests.py -t snowflake

...

SUCCESS: ['string_column_anomalies'] in list ['string_column_anomalies']
SUCCESS: ['min_length'] in list ['min_length']
SUCCESS: ['relationships'] in list ['relationships']
2022-05-30 15:24:18,525 — monitor.dbt_runner — INFO — Running dbt --log-format json run-operation validate_dbt_artifacts --project-dir  --profiles-dir /home/ec/.dbt --target snowflake (this might take a while)
SUCCESS: ANY_TYPE_COLUMN_ANOMALIES in list ['data_monitoring_metrics', 'column_changes', 'schema_tables_snapshot', 'table_changes', 'current_schema_columns', 'previous_schema_columns', 'current_schema_tables', 'filtered_information_schema_columns', 'monitors_runs', 'alerts_schema_changes', 'numeric_column_anomalies', 'string_column_anomalies', 'test_run_results', 'alerts_data_monitoring', 'anomaly_threshold_sensitivity', 'filtered_information_schema_tables', 'any_type_column_anomalies', 'metrics_anomaly_score', 'schema_columns_snapshot', 'alerts_dbt_tests', 'previous_schema_tables', 'model_run_results']
SUCCESS: NUMERIC_COLUMN_ANOMALIES in list ['data_monitoring_metrics', 'column_changes', 'schema_tables_snapshot', 'table_changes', 'current_schema_columns', 'previous_schema_columns', 'current_schema_tables', 'filtered_information_schema_columns', 'monitors_runs', 'alerts_schema_changes', 'numeric_column_anomalies', 'string_column_anomalies', 'test_run_results', 'alerts_data_monitoring', 'anomaly_threshold_sensitivity', 'filtered_information_schema_tables', 'any_type_column_anomalies', 'metrics_anomaly_score', 'schema_columns_snapshot', 'alerts_dbt_tests', 'previous_schema_tables', 'model_run_results']
SUCCESS: STRING_COLUMN_ANOMALIES in list ['data_monitoring_metrics', 'column_changes', 'schema_tables_snapshot', 'table_changes', 'current_schema_columns', 'previous_schema_columns', 'current_schema_tables', 'filtered_information_schema_columns', 'monitors_runs', 'alerts_schema_changes', 'numeric_column_anomalies', 'string_column_anomalies', 'test_run_results', 'alerts_data_monitoring', 'anomaly_threshold_sensitivity', 'filtered_information_schema_tables', 'any_type_column_anomalies', 'metrics_anomaly_score', 'schema_columns_snapshot', 'alerts_dbt_tests', 'previous_schema_tables', 'model_run_results']
SUCCESS: ANY_TYPE_COLUMN_ANOMALIES_TRAINING in list ['dbt_run_results', 'dbt_models', 'dbt_tests', 'dbt_sources', 'dbt_metrics', 'any_type_column_anomalies_training', 'string_column_anomalies_training', 'stats_players', 'stats_team', 'any_type_column_anomalies_validation', 'groups', 'dbt_exposures']
SUCCESS: STRING_COLUMN_ANOMALIES_TRAINING in list ['dbt_run_results', 'dbt_models', 'dbt_tests', 'dbt_sources', 'dbt_metrics', 'any_type_column_anomalies_training', 'string_column_anomalies_training', 'stats_players', 'stats_team', 'any_type_column_anomalies_validation', 'groups', 'dbt_exposures']
SUCCESS: ANY_TYPE_COLUMN_ANOMALIES_VALIDATION in list ['dbt_run_results', 'dbt_models', 'dbt_tests', 'dbt_sources', 'dbt_metrics', 'any_type_column_anomalies_training', 'string_column_anomalies_training', 'stats_players', 'stats_team', 'any_type_column_anomalies_validation', 'groups', 'dbt_exposures']
SUCCESS: ['alerts_dbt_tests', 'alerts_schema_changes', 'data_monitoring_metrics', 'filtered_information_schema_columns', 'filtered_information_schema_tables', 'model_run_results', 'test_run_results', 'numeric_column_anomalies', 'string_column_anomalies', 'metrics_anomaly_score', 'monitors_runs', 'schema_columns_snapshot', 'schema_tables_snapshot', 'column_changes', 'elementary_source_table_anomalies_training_any_type_column_anomalies_training_', 'elementary_table_anomalies_any_type_column_anomalies_row_count', 'elementary_table_anomalies_numeric_column_anomalies_', 'elementary_table_anomalies_string_column_anomalies_freshness', 'elementary_column_anomalies_string_column_anomalies_max_length', 'elementary_column_anomalies_string_column_anomalies_min_length__max_length__missing_count__min_length', 'elementary_column_anomalies_string_column_anomalies_missing_percent', 'elementary_column_anomalies_numeric_column_anomalies_average__average', 'elementary_column_anomalies_numeric_column_anomalies_max__average', 'elementary_column_anomalies_numeric_column_anomalies_max__min', 'elementary_column_anomalies_numeric_column_anomalies_min__max', 'elementary_column_anomalies_numeric_column_anomalies_min__min', 'elementary_column_anomalies_numeric_column_anomalies_standard_deviation', 'elementary_column_anomalies_numeric_column_anomalies_zero_count', 'elementary_all_columns_anomalies_any_type_column_anomalies__column1_column2_column3_column4_column5_column6_column7_column8_column9_column10_column11_column12_column13_column14_column15_column16_column17_', 'elementary_source_schema_changes_schema_changes_data_groups_', 'anomaly_threshold_sensitivity', 'elementary_column_anomalies_numeric_column_anomalies_max__max', 'elementary_column_anomalies_numeric_column_anomalies_updated_at', 'elementary_schema_changes_string_column_anomalies_', 'alerts_data_monitoring', 'table_changes', 'elementary_column_anomalies_string_column_anomalies_average_length__null_count__average_length', 'elementary_column_anomalies_numeric_column_anomalies_average__max', 'elementary_column_anomalies_numeric_column_anomalies_zero_percent', 'any_type_column_anomalies', 'elementary_source_table_anomalies_training_string_column_anomalies_training_', 'elementary_column_anomalies_string_column_anomalies_missing_count', 'elementary_column_anomalies_numeric_column_anomalies_average__min', 'elementary_column_anomalies_numeric_column_anomalies_min__average', 'elementary_column_anomalies_numeric_column_anomalies_variance', 'elementary_source_schema_changes_schema_changes_data_stats_players_', 'relationships_string_column_anomalies_min_length__max_length__source_training_string_column_anomalies_training_', 'elementary_schema_changes_numeric_column_anomalies_', 'elementary_column_anomalies_string_column_anomalies_updated_at', 'elementary_source_schema_changes_schema_changes_data_stats_team_'] in list ['data_monitoring_metrics', 'column_changes', 'schema_tables_snapshot', 'table_changes', 'current_schema_columns', 'previous_schema_columns', 'current_schema_tables', 'filtered_information_schema_columns', 'monitors_runs', 'alerts_schema_changes', 'numeric_column_anomalies', 'string_column_anomalies', 'test_run_results', 'alerts_data_monitoring', 'anomaly_threshold_sensitivity', 'filtered_information_schema_tables', 'any_type_column_anomalies', 'metrics_anomaly_score', 'schema_columns_snapshot', 'alerts_dbt_tests', 'previous_schema_tables', 'model_run_results', 'elementary_table_anomalies_any_type_column_anomalies_row_count', 'elementary_all_columns_anomalies_any_type_column_anomalies__column1_column2_column3_column4_column5_column6_column7_column8_column9_column10_column11_column12_column13_column14_column15_column16_column17_', 'elementary_column_anomalies_numeric_column_anomalies_min__min', 'elementary_column_anomalies_numeric_column_anomalies_max__min', 'elementary_column_anomalies_numeric_column_anomalies_min__max', 'elementary_column_anomalies_numeric_column_anomalies_max__max', 'elementary_column_anomalies_numeric_column_anomalies_min__average', 'elementary_column_anomalies_numeric_column_anomalies_average__average', 'elementary_column_anomalies_numeric_column_anomalies_zero_percent', 'elementary_column_anomalies_numeric_column_anomalies_variance', 'elementary_column_anomalies_numeric_column_anomalies_standard_deviation', 'elementary_table_anomalies_numeric_column_anomalies_', 'elementary_column_anomalies_string_column_anomalies_min_length__max_length__missing_count__min_length', 'elementary_column_anomalies_string_column_anomalies_missing_count', 'elementary_column_anomalies_string_column_anomalies_missing_percent', 'elementary_column_anomalies_string_column_anomalies_updated_at', 'elementary_all_columns_anomalies_string_column_anomalies_', 'elementary_source_table_anomalies_training_any_type_column_anomalies_training_', 'elementary_source_schema_changes_schema_changes_data_groups_', 'elementary_source_schema_changes_schema_changes_data_stats_players_', 'elementary_column_anomalies_numeric_column_anomalies_average__max', 'elementary_all_columns_anomalies_numeric_column_anomalies_average_length__null_count', 'relationships_string_column_anomalies_min_length__max_length__source_training_string_column_anomalies_training_', 'elementary_source_all_columns_anomalies_validation_any_type_column_anomalies_validation_', 'elementary_column_anomalies_numeric_column_anomalies_average__min', 'elementary_column_anomalies_numeric_column_anomalies_max__average', 'elementary_schema_changes_numeric_column_anomalies_', 'elementary_column_anomalies_string_column_anomalies_average_length__null_count__average_length', 'elementary_table_anomalies_string_column_anomalies_freshness', 'elementary_schema_changes_string_column_anomalies_', 'elementary_column_anomalies_numeric_column_anomalies_updated_at', 'elementary_source_schema_changes_schema_changes_data_stats_team_', 'elementary_column_anomalies_numeric_column_anomalies_zero_count', 'elementary_column_anomalies_string_column_anomalies_max_length', 'elementary_source_table_anomalies_training_string_column_anomalies_training_']

snowflake results

snowflake results

Table test results
SUCCESS: ['STRING_COLUMN_ANOMALIES_TRAINING', 'STRING_COLUMN_ANOMALIES', 'NUMERIC_COLUMN_ANOMALIES'] in list ['string_column_anomalies', 'numeric_column_anomalies', 'string_column_anomalies_training']
SUCCESS: ['STRING_COLUMN_ANOMALIES_TRAINING', 'ANY_TYPE_COLUMN_ANOMALIES', 'NUMERIC_COLUMN_ANOMALIES'] in list ['any_type_column_anomalies', 'numeric_column_anomalies', 'string_column_anomalies_training']

String columns test results
SUCCESS: ['AVERAGE_LENGTH', 'MAX_LENGTH', 'MIN_LENGTH', 'MISSING_COUNT', 'MISSING_PERCENT'] in list ['min_length', 'max_length', 'average_length', 'missing_count', 'missing_percent']

Numeric columns test results
SUCCESS: ['AVERAGE', 'MIN', 'STANDARD_DEVIATION', 'VARIANCE', 'ZERO_COUNT', 'ZERO_PERCENT', 'MAX'] in list ['min', 'max', 'zero_count', 'zero_percent', 'average', 'standard_deviation', 'variance']

Any type columns test results
SUCCESS: ['NULL_COUNT_INT', 'NULL_COUNT_STR', 'NULL_PERCENT_STR', 'NULL_PERCENT_INT', 'NULL_PERCENT_FLOAT', 'NULL_PERCENT_BOOL', 'NULL_COUNT_FLOAT', 'NULL_COUNT_BOOL'] in list ['null_count_str', 'null_percent_str', 'null_count_float', 'null_percent_float', 'null_count_int', 'null_percent_int', 'null_count_bool', 'null_percent_bool']

Schema changes test results
SUCCESS: all expected schema changes were found - {'group_a': 'column_removed', 'offsides': 'column_removed', 'key_crosses': 'column_added', 'red_cards': 'column_added'}

Regular test results
SUCCESS: ['string_column_anomalies'] in list ['string_column_anomalies']
SUCCESS: ['min_length'] in list ['min_length']
SUCCESS: ['relationships'] in list ['relationships']

dbt artifacts results
SUCCESS: ANY_TYPE_COLUMN_ANOMALIES in list ['data_monitoring_metrics', 'column_changes', 'schema_tables_snapshot', 'table_changes', 'current_schema_columns', 'previous_schema_columns', 'current_schema_tables', 'filtered_information_schema_columns', 'monitors_runs', 'alerts_schema_changes', 'numeric_column_anomalies', 'string_column_anomalies', 'test_run_results', 'alerts_data_monitoring', 'anomaly_threshold_sensitivity', 'filtered_information_schema_tables', 'any_type_column_anomalies', 'metrics_anomaly_score', 'schema_columns_snapshot', 'alerts_dbt_tests', 'previous_schema_tables', 'model_run_results']
SUCCESS: NUMERIC_COLUMN_ANOMALIES in list ['data_monitoring_metrics', 'column_changes', 'schema_tables_snapshot', 'table_changes', 'current_schema_columns', 'previous_schema_columns', 'current_schema_tables', 'filtered_information_schema_columns', 'monitors_runs', 'alerts_schema_changes', 'numeric_column_anomalies', 'string_column_anomalies', 'test_run_results', 'alerts_data_monitoring', 'anomaly_threshold_sensitivity', 'filtered_information_schema_tables', 'any_type_column_anomalies', 'metrics_anomaly_score', 'schema_columns_snapshot', 'alerts_dbt_tests', 'previous_schema_tables', 'model_run_results']
SUCCESS: STRING_COLUMN_ANOMALIES in list ['data_monitoring_metrics', 'column_changes', 'schema_tables_snapshot', 'table_changes', 'current_schema_columns', 'previous_schema_columns', 'current_schema_tables', 'filtered_information_schema_columns', 'monitors_runs', 'alerts_schema_changes', 'numeric_column_anomalies', 'string_column_anomalies', 'test_run_results', 'alerts_data_monitoring', 'anomaly_threshold_sensitivity', 'filtered_information_schema_tables', 'any_type_column_anomalies', 'metrics_anomaly_score', 'schema_columns_snapshot', 'alerts_dbt_tests', 'previous_schema_tables', 'model_run_results']
SUCCESS: ANY_TYPE_COLUMN_ANOMALIES_TRAINING in list ['dbt_run_results', 'dbt_models', 'dbt_tests', 'dbt_sources', 'dbt_metrics', 'any_type_column_anomalies_training', 'string_column_anomalies_training', 'stats_players', 'stats_team', 'any_type_column_anomalies_validation', 'groups', 'dbt_exposures']
SUCCESS: STRING_COLUMN_ANOMALIES_TRAINING in list ['dbt_run_results', 'dbt_models', 'dbt_tests', 'dbt_sources', 'dbt_metrics', 'any_type_column_anomalies_training', 'string_column_anomalies_training', 'stats_players', 'stats_team', 'any_type_column_anomalies_validation', 'groups', 'dbt_exposures']
SUCCESS: ANY_TYPE_COLUMN_ANOMALIES_VALIDATION in list ['dbt_run_results', 'dbt_models', 'dbt_tests', 'dbt_sources', 'dbt_metrics', 'any_type_column_anomalies_training', 'string_column_anomalies_training', 'stats_players', 'stats_team', 'any_type_column_anomalies_validation', 'groups', 'dbt_exposures']
SUCCESS: ['alerts_dbt_tests', 'alerts_schema_changes', 'data_monitoring_metrics', 'filtered_information_schema_columns', 'filtered_information_schema_tables', 'model_run_results', 'test_run_results', 'numeric_column_anomalies', 'string_column_anomalies', 'metrics_anomaly_score', 'monitors_runs', 'schema_columns_snapshot', 'schema_tables_snapshot', 'column_changes', 'elementary_source_table_anomalies_training_any_type_column_anomalies_training_', 'elementary_table_anomalies_any_type_column_anomalies_row_count', 'elementary_table_anomalies_numeric_column_anomalies_', 'elementary_table_anomalies_string_column_anomalies_freshness', 'elementary_column_anomalies_string_column_anomalies_max_length', 'elementary_column_anomalies_string_column_anomalies_min_length__max_length__missing_count__min_length', 'elementary_column_anomalies_string_column_anomalies_missing_percent', 'elementary_column_anomalies_numeric_column_anomalies_average__average', 'elementary_column_anomalies_numeric_column_anomalies_max__average', 'elementary_column_anomalies_numeric_column_anomalies_max__min', 'elementary_column_anomalies_numeric_column_anomalies_min__max', 'elementary_column_anomalies_numeric_column_anomalies_min__min', 'elementary_column_anomalies_numeric_column_anomalies_standard_deviation', 'elementary_column_anomalies_numeric_column_anomalies_zero_count', 'elementary_all_columns_anomalies_any_type_column_anomalies__column1_column2_column3_column4_column5_column6_column7_column8_column9_column10_column11_column12_column13_column14_column15_column16_column17_', 'elementary_source_schema_changes_schema_changes_data_groups_', 'anomaly_threshold_sensitivity', 'elementary_column_anomalies_numeric_column_anomalies_max__max', 'elementary_column_anomalies_numeric_column_anomalies_updated_at', 'elementary_schema_changes_string_column_anomalies_', 'alerts_data_monitoring', 'table_changes', 'elementary_column_anomalies_string_column_anomalies_average_length__null_count__average_length', 'elementary_column_anomalies_numeric_column_anomalies_average__max', 'elementary_column_anomalies_numeric_column_anomalies_zero_percent', 'any_type_column_anomalies', 'elementary_source_table_anomalies_training_string_column_anomalies_training_', 'elementary_column_anomalies_string_column_anomalies_missing_count', 'elementary_column_anomalies_numeric_column_anomalies_average__min', 'elementary_column_anomalies_numeric_column_anomalies_min__average', 'elementary_column_anomalies_numeric_column_anomalies_variance', 'elementary_source_schema_changes_schema_changes_data_stats_players_', 'relationships_string_column_anomalies_min_length__max_length__source_training_string_column_anomalies_training_', 'elementary_schema_changes_numeric_column_anomalies_', 'elementary_column_anomalies_string_column_anomalies_updated_at', 'elementary_source_schema_changes_schema_changes_data_stats_team_'] in list ['data_monitoring_metrics', 'column_changes', 'schema_tables_snapshot', 'table_changes', 'current_schema_columns', 'previous_schema_columns', 'current_schema_tables', 'filtered_information_schema_columns', 'monitors_runs', 'alerts_schema_changes', 'numeric_column_anomalies', 'string_column_anomalies', 'test_run_results', 'alerts_data_monitoring', 'anomaly_threshold_sensitivity', 'filtered_information_schema_tables', 'any_type_column_anomalies', 'metrics_anomaly_score', 'schema_columns_snapshot', 'alerts_dbt_tests', 'previous_schema_tables', 'model_run_results', 'elementary_table_anomalies_any_type_column_anomalies_row_count', 'elementary_all_columns_anomalies_any_type_column_anomalies__column1_column2_column3_column4_column5_column6_column7_column8_column9_column10_column11_column12_column13_column14_column15_column16_column17_', 'elementary_column_anomalies_numeric_column_anomalies_min__min', 'elementary_column_anomalies_numeric_column_anomalies_max__min', 'elementary_column_anomalies_numeric_column_anomalies_min__max', 'elementary_column_anomalies_numeric_column_anomalies_max__max', 'elementary_column_anomalies_numeric_column_anomalies_min__average', 'elementary_column_anomalies_numeric_column_anomalies_average__average', 'elementary_column_anomalies_numeric_column_anomalies_zero_percent', 'elementary_column_anomalies_numeric_column_anomalies_variance', 'elementary_column_anomalies_numeric_column_anomalies_standard_deviation', 'elementary_table_anomalies_numeric_column_anomalies_', 'elementary_column_anomalies_string_column_anomalies_min_length__max_length__missing_count__min_length', 'elementary_column_anomalies_string_column_anomalies_missing_count', 'elementary_column_anomalies_string_column_anomalies_missing_percent', 'elementary_column_anomalies_string_column_anomalies_updated_at', 'elementary_all_columns_anomalies_string_column_anomalies_', 'elementary_source_table_anomalies_training_any_type_column_anomalies_training_', 'elementary_source_schema_changes_schema_changes_data_groups_', 'elementary_source_schema_changes_schema_changes_data_stats_players_', 'elementary_column_anomalies_numeric_column_anomalies_average__max', 'elementary_all_columns_anomalies_numeric_column_anomalies_average_length__null_count', 'relationships_string_column_anomalies_min_length__max_length__source_training_string_column_anomalies_training_', 'elementary_source_all_columns_anomalies_validation_any_type_column_anomalies_validation_', 'elementary_column_anomalies_numeric_column_anomalies_average__min', 'elementary_column_anomalies_numeric_column_anomalies_max__average', 'elementary_schema_changes_numeric_column_anomalies_', 'elementary_column_anomalies_string_column_anomalies_average_length__null_count__average_length', 'elementary_table_anomalies_string_column_anomalies_freshness', 'elementary_schema_changes_string_column_anomalies_', 'elementary_column_anomalies_numeric_column_anomalies_updated_at', 'elementary_source_schema_changes_schema_changes_data_stats_team_', 'elementary_column_anomalies_numeric_column_anomalies_zero_count', 'elementary_column_anomalies_string_column_anomalies_max_length', 'elementary_source_table_anomalies_training_string_column_anomalies_training_']
```

oravi commented:
Great work @Ecalzo ! merging it now
